test_log/src/main/java/org/corfudb/test_log/TestLog.java

```diff
@@ -0,0 +1,289 @@
+            record_buffer.flip();
+            LogEntry le = LogEntry.parseFrom(record_buffer.array());
+            HashCode hc = hf.newHasher()
+                    .putBytes(record_buffer.array())
+                    .hash();
+            if (hc.asInt() != md.getChecksum()) {
+                System.out.println("Checksum ERROR");
+            }
+            if (display) {
+                System.out.format("Computed CRC32C checksum %08x\n", hc.asInt());
+                System.out.format("Log Entry streams (%d):  ", le.getStreamsCount());
+                for (int i = 0; i < le.getStreamsCount(); i++) {
+                    System.out.print(le.getStreams(i) + " ");
+                }
+                System.out.format("\n");
+            }
+            if (display_data && le.hasData()) {
+                String bstr = new String();
+            	ByteString dbuff = le.getData();
+                for (int i=0; i < dbuff.size(); i++) {
+                    byte c = dbuff.byteAt(i);
+                    if (Character.isLetterOrDigit(c)) {
+                        bstr += (char) c;
+                    } else {
+                        bstr += String.format("\\x%02x", c);
+                    }
+                }
+                System.out.format("Data:\n%s\n", bstr);
+            }
+            long global_addr = le.hasGlobalAddress() ? le.getGlobalAddress() : -1;
+            if (global_addr >= 0) {
+                if (display) {
+                    System.out.format("Global address: %d\n", global_addr);
+                }
+                if (global_addr == op.getGlobalAddress()) {
+                    trigger = true;
+                }
+            }
+            if (le.getLogicalAddressesCount() > 0) {
+                java.util.Map<String, Long> map = le.getLogicalAddressesMap();
+            }
+            if (display && le.hasRank()) {
+                System.out.format("Rank: %d\n", le.getRank());
+            }
+            if (display && le.hasCommit()) {
+                System.out.format("Commit: ");
+                System.out.println(le.getCommit());
+            }
+            if (display && le.hasDataType()) {
+                DataType dt = le.getDataType();
+                switch (dt) {
+                    case DATA:
+                        System.out.println("DataType: DATA");
+                        break;
+                    case EMPTY:
+                        System.out.println("DataType: EMPTY");
+                        break;
+                    case HOLE:
+                        System.out.println("DataType: HOLE");
+                        break;
+                    case TRIMMED:
+                        System.out.println("DataType: TRIMMED");
+                        break;
+                    default:
+                        System.out.println("UNKNOWN DataType");
+                        break;
+                }
+            }
+            replace_entry |= (trigger &&
```

Comment:
What is the trigger ? Please provide comments. I am guessing some of these variables need to be in smaller scopes once you have broken this method into smaller more cohesive ones.